projects/Objects/HomogenousTree/src/com/valentin/HTree.java

```diff
@@ -0,0 +1,144 @@
+package com.valentin;
+
+
+import java.util.Vector;
+
+public class HTree {
+    Vector<Node> nodesVector=new Vector();
+    Node rootNode=new Node();
+    boolean flagContains;
+
+    public HTree(){}
+
+    /**
+     *
+     * @param object
+     * @param value
+     */
+    public void add(Object object,int value){
+        Node currentNode=new Node(object,value);
+        if (!nodesVector.isEmpty()){
+            if (contains(value)){
+                System.out.println("this value is already used");
```

Comment:
System.out in production code is not a good idea. 

You can improve it by making add method to return boolean value which to indicate whether element was added or not. So in the main app you will have:
```java
if (!tree.add(10, "MyObject")) {
   // now users know that 10 was already added
   System.out.printf("The key '%d' already exists and cannot be added twice", 10)
}
```